src/vs/workbench/api/electron-browser/mainThreadLanguages.ts

```diff
@@ -22,4 +26,12 @@ export class MainThreadLanguages extends MainThreadLanguagesShape {
 	$getLanguages(): TPromise<string[]> {
 		return TPromise.as(this._modeService.getRegisteredModes());
 	}
+
+	$getLanguage(resource: URI, position?: IPosition): TPromise<string> {
+		const model = this.modelService.getModel(resource);
+		if (model) {
+			return TPromise.as(position ? this._modeService.getLanguageIdentifier(model.getLanguageIdAtPosition(position.lineNumber, position.column)).language : model.getLanguageIdentifier().language);
+		}
+		return TPromise.as(this._modeService.getModeIdByFilenameOrFirstLine(resource.fsPath));
```

Comment:
That won't work because to really know the mode you need to peek into the file. Think of an executable file `app` with a first line `#!/bin/node`. I see that in other places we use that method equally sloppy but for the API there are  different requirements. Unsure how the workbench does it, @bpasero will know